apps/meteor/app/federation-v2/server/infrastructure/rocket-chat/adapters/Room.ts

```diff
@@ -68,39 +88,17 @@ export class RocketChatRoomAdapter {
 		);
 	}
 
-	public async updateRoomType(federatedRoom: FederatedRoom): Promise<void> {
-		await Rooms.update({ _id: federatedRoom.internalReference._id }, { $set: { t: federatedRoom.internalReference.t } });
-		await Subscriptions.update(
-			{ rid: federatedRoom.internalReference._id },
-			{ $set: { t: federatedRoom.internalReference.t } },
-			{ multi: true },
-		);
-	}
-
-	public async updateRoomName(federatedRoom: FederatedRoom): Promise<void> {
-		await Rooms.update(
-			{ _id: federatedRoom.internalReference._id },
-			{ $set: { name: federatedRoom.internalReference.name, fname: federatedRoom.internalReference.fname } },
-		);
-		await Subscriptions.update(
-			{ rid: federatedRoom.internalReference._id },
-			{ $set: { name: federatedRoom.internalReference.name, fname: federatedRoom.internalReference.fname } },
-			{ multi: true },
-		);
-	}
-
-	public async updateRoomTopic(federatedRoom: FederatedRoom): Promise<void> {
-		await Rooms.update(
-			{ _id: federatedRoom.internalReference._id },
-			{ $set: { description: federatedRoom.internalReference.description } },
-		);
-	}
-
 	private createFederatedRoomInstance(externalRoomId: string, room: IRoom): FederatedRoom {
 		const federatedRoom = FederatedRoom.build();
 		federatedRoom.externalId = externalRoomId;
 		federatedRoom.internalReference = room;
 
 		return federatedRoom;
 	}
+
+	public async isUserAlreadyJoined(internalRoomId: string, internalUserId: string): Promise<boolean> {
+		const subscription = await Subscriptions.findOneByRoomIdAndUserId(internalRoomId, internalUserId);
```

Comment:
no need to fetch all fields
```suggestion
		const subscription = await Subscriptions.findOneByRoomIdAndUserId(internalRoomId, internalUserId, { projection: { _id: 1 } });
```